# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmModule.cpp
# Commit: a7ad7e2b02c1
# Full Hash: a7ad7e2b02c11538c1f27efb67a48a4fb823ba8e
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-18 09:24:52
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Generalize table/element code for future reference-types. r=lth
#   
#   wasm::Table uses wasm::TableKind to determine the element type of the table,
#   which makes it difficult to extend tables to support more reference types. This
#   commit drops TableKind in favor of directly storing the element RefType, along
# ==============================================================================

diff -r 538920d2d777 -r a7ad7e2b02c1 js/src/wasm/WasmModule.cpp
--- a/js/src/wasm/WasmModule.cpp	Tue Aug 18 00:04:37 2020 +0000
+++ b/js/src/wasm/WasmModule.cpp	Tue Aug 18 00:45:07 2020 +0000
@@ -801,7 +801,7 @@
     RootedObject proto(
         cx, &cx->global()->getPrototype(JSProto_WasmTable).toObject());
     tableObj.set(WasmTableObject::create(cx, td.initialLength, td.maximumLength,
-                                         td.kind, proto));
+                                         td.elemType, proto));
     if (!tableObj) {
       return false;
     }